internal/tpm/tpm.go

```diff
@@ -51,3 +91,34 @@ func (t *TPM) GetPCRValues(measurements map[string]string) error {
 	}
 	return nil
 }
+
+// The local attestation key (LAK) is an asymmetric key that persists for the device's lifecycle (but not lifetime) and can be zeroized if needed when the device transfers ownership. (The IAK by contrast persists for the device's lifetime across uses and owners.) This key can only be used to sign TPM-internal data, ex. attestations. This is considered a Restricted signing key by the TPM.
+// Key attributes:
+// Restricted: yes
+// Sign: yes
+// Decrypt: no
+// FixedTPM: yes (cannot migrate or be duplicated)
+// SensitiveDataOrigin: yes (was created in the TPM)
+func (t *TPM) CreateLAK() (*client.Key, error) {
+	// AttestationKeyECC generates and loads a key from AKTemplateECC in the Owner (aka 'Storage') hierarchy.
+	return client.AttestationKeyECC(t.channel)
+}
+
+func (t *TPM) GetAttestation(nonce []byte, ak *client.Key) (*pbattest.Attestation, error) {
+	// TODO - may want to use CertChainFetcher in the AttestOpts in the future
+	// see https://pkg.go.dev/github.com/google/go-tpm-tools/client#AttestOpts
+
+	if len(nonce) < MinNonceLength {
+		return nil, fmt.Errorf("nonce does not meet minimum length of %d bytes", MinNonceLength)
+	}
+
+	return ak.Attest(client.AttestOpts{Nonce: nonce})
+}
+
+func (t *TPM) GetQuote(nonce []byte, ak *client.Key, pcr_selection *tpm2.PCRSelection) (*pbtpm.Quote, error) {
+	if len(nonce) < MinNonceLength {
+		return nil, fmt.Errorf("nonce does not meet minimum length of %d bytes", MinNonceLength)
+	}
+
+	return ak.Quote(*pcr_selection, nonce)
```

Comment:
question: Usual go patterns would guarantee these params would not be nil, are we certain of that here.

Reply:
This is a good catch, I always forget you have to do this check manually in Go.

Reply:
well the pattern should ensure they were nil checked before passed to the method just making sure that is true.

Reply:
I wouldn't assume it gets checked, so I added the check.